File: Leetcode/1310/1310.go
func rangeXorSum(pre []int, l, r int) int {
	if l == 1 {
		return pre[r]
	}
	return pre[l-1] ^ pre[r]
}
func xorQueries(arr []int, queries [][]int) []int {
	pre := make([]int, len(arr)+1)
	for i := 1; i <= len(arr); i++ {
		pre[i] = pre[i-1] ^ arr[i-1]
	}
	ans := make([]int, len(queries))
	for i := 0; i < len(queries); i++ {
		ans[i] = rangeXorSum(pre, queries[i][0]+1, queries[i][1]+1)
	}
	return ans
}

File: Leetcode/1310/1310.cpp
class Solution {
public:
    int rangeXorSum(vector<int>& pre, int l, int r) {
        if (l == 1) {
            return pre[r];
        }
        return pre[l-1]^pre[r];
    }
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        int n = (int)arr.size();
        vector<int> pre(n+1);
        for (int i = 1; i <= n; ++i) {
            pre[i] = pre[i-1]^arr[i-1];
        }
        int q = (int)queries.size();
        vector<int> ans(q);
        for (int i = 0; i < q; ++i) {
            ans[i] = rangeXorSum(pre, queries[i][0]+1, queries[i][1]+1);
        }
        return ans;
    }
};